ScenarioManager/atomic_scenario_behavior.py

```diff
@@ -666,3 +668,48 @@ def update(self):
         self._vehicle.apply_control(self._control)
 
         return new_status
+
+
+class BasicAgentBehavior(AtomicBehavior):
```

Comment:
The behaviour works fine for me. I tested scenario 9 to use this behaviour to execute turn. It behaves as expected.

Since we are trying to reduce location coupled parameters, is it ok to be using target location ? May be it can take a list of directives (left, right, straight, etc.) to execute like how the basic agent consumes [global route plan](https://github.com/carla-simulator/carla/pull/1174) ?

Reply:
As soon as your PR is merged, we can update the behavior to use turn information. I would see this as another PR.